packages/dds/sequence/src/intervalCollection.ts

```diff
@@ -1572,8 +1571,9 @@ export class IntervalCollection
 			setSlideOnRemove(interval.end);
 		}
 
-		const needsStartUpdate = newStart !== undefined && !hasPendingChange;
-		const needsEndUpdate = newEnd !== undefined && !hasPendingChange;
+		const needsStartUpdate =
```

Comment:
the code here was very brittle and dependent on a quirk of `getSlideToSegoff` that would sometimes return a malformed result with only an `{offset:<something>}`, which then would have `getSlideToSegment` return the same malformed result and then in `createPositionReferenceFromSegoff` if would become a `detachedReferencePosition`

the new check is much more semantically correct, and allowed the clean up of the quicks in the prior implementation .